src/Servers/Kestrel/Core/src/ITlsConfigurationLoader.cs

```diff
@@ -0,0 +1,24 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal;
+using Microsoft.AspNetCore.Server.Kestrel.Https;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core;
+
+internal interface ITlsConfigurationLoader
```

Comment:
This could be flattened into `IHttpsConfigurationHelper`, if that seems more consistent.

Reply:
I'm curious why made them separate interfaces? Is it logic that was in two locations?

`UseHttps` method on each interface seems weird.

Reply:
It's mostly an artifact of the many revisions this change has gone through.  The `UseHttps` method on the interface is badly named, but I couldn't think of a better one.  It is substantially different from the one on `IHttpsConfigurationHelper`.

Reply:
The class still encapsulates some state nicely, but I've merged the interfaces.